aip/general/0122.md

```diff
@@ -17,7 +17,7 @@ resources.
 
 ## Guidance
 
-All resource names defined by an API **must** be unique within that API. (See
+All resource names defined by an API **must** be unique and unambiguous within that API. (See
```

Comment:
This sounds somewhat redundant to me - how could it be unique but ambiguous?

Reply:
Thanks for the quick review.

I agree this doesn't help much. I reverted this change.

What I'd like to say is that even names of static collections, like locations (which should always themselves be resources?), must be unique within an API/service. Do we have specific terminology for those? What I want to clarify is that the same collection names should not return different collections through different API endpoints.

Probably a new paragraph is needed somewhere.

Reply:
> What I want to clarify is that the same collection names should not return different collections through different API endpoints.

I suspect an internal doc with "what we want to avoid" would be helpful (at least for me) in understanding the concern. For example

- Are we worried about different endpoints returning different *types* for the same collection name?
- Are we worried about different endpoints returning different *sets of resources* for the same collection name? (e.g. a regionalized endpoint that only ever returned items in its region)
- Something else I haven't thought about?

(And is this within a single API version? I'd expect /v1/databases/xyz and /v2/databases/xyz to be different messages, for example... whether or not they return the same set of resources.)

I don't have any suggestions for improving the PR at this point, and others may well already be more in tune with the concern and be able to review it usefully.

Reply:
Different sets of resources for the same collection name within the same API version. 

I could punt that change to a separate PR.